src/traits.rs

```diff
@@ -119,6 +119,45 @@ pub trait Context {
         hostcalls::cancel_grpc_call(token_id)
     }
 
+    fn create_grpc_stream(
+        &self,
+        cluster_name: &str,
+        service_name: &str,
+        method_name: &str,
+        initial_metadata: Vec<(&str, &[u8])>,
+    ) -> Result<u32, Status> {
+        hostcalls::create_grpc_stream(cluster_name, service_name, method_name, initial_metadata)
+    }
+
+    fn grpc_stream_send(
+        &self,
+        token_id: u32,
+        message: Option<&[u8]>,
+        end_stream: bool,
+    ) -> Result<(), Status> {
+        hostcalls::grpc_stream_send(token_id, message, end_stream)
+    }
+
+    fn grpc_stream_close(&self, token_id: u32) -> Result<(), Status> {
+        hostcalls::grpc_stream_close(token_id)
+    }
+
+    fn on_grpc_stream_receive_initial_metadata(&mut self, _token_id: u32, _headers: u32) {}
+
+    fn on_grpc_stream_receive_trailing_metadata(&mut self, _token_id: u32, _trailers: u32) {}
```

Comment:
Nit: drop `receive`, it's doesn't add any value here, so `on_grpc_stream_trailing_metadata`.

Also, replace `_trailers` with `_num_elements`